dts/bindings/can/nxp,s32-canxl.yaml

```diff
@@ -0,0 +1,27 @@
+# Copyright 2022 NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP S32 CANXL controller
+
+compatible: "nxp,s32-canxl"
```

Comment:
As of my knowledge, there’s no other device ported to Zephyr that has this peripheral, except from the S32 family. On the other hand, it wouldn’t be the only binding that have a family prefix but it is also used in SoCs of another families (I’ve seen this for some kinetis and imx peripherals). I agree that if we can avoid that from the beginning it would be cleaner. But so far we don’t have this information.